lib/src/doc/process.rs

```diff
@@ -0,0 +1,78 @@
+use crate::ctx::Context;
+use crate::dbs::Statement;
+use crate::dbs::Workable;
+use crate::dbs::{Operable, Transaction};
+use crate::dbs::{Options, Processed};
+use crate::doc::Document;
+use crate::err::Error;
+use crate::sql::value::Value;
+
+impl<'a> Document<'a> {
+	#[allow(dead_code)]
+	pub(crate) async fn process(
+		ctx: &Context<'_>,
+		opt: &Options,
+		txn: &Transaction,
+		stm: &Statement<'_>,
+		mut pro: Processed,
+	) -> Result<Value, Error> {
+		// Loop over maximum two times
+		for _ in 0..2 {
```

Comment:
I understand that we need to retry because of the RetryWithId error, but why is 2 the right value?
If my understanding is correct, we need that in case we are using `parallel`.
Then depending on the contention, we may need a higher value.
We may also add random temporisation when this error occurs, so it gives a chance for one process to win and let every concurrent statement progress.